HW-1 Upbit API를 활용한 코인 마켓 리스트 조회 기능 구현

BeokBeok opened the pull request:
1주차 과제 PR 요청 드립니다.

Commit:
HW-1 메인 Layout 구성 및 리소스 추가

Commit:
HW-1 마켓 정보 조회 및 원화 단위 필터링 적용

- 테스트 버튼을 추가 및 동작 확인 완료
(추후 테스트 버튼 제거)

Commit:
HW-1 Ticker 를 통한 마켓 정보 조회 기능 구현

- MarketAll 클래스를 Ticker 클래스에 통합
- 테스트 버튼 제거

Commit:
HW-1 KRW 기준으로 마켓 시세 보여주기 기능 구현

sjjeong commented:
PR이름을 수정해주세요
ex HW-1 과제

BeokBeok commented:
> PR이름을 수정해주세요
> ex HW-1 과제

변경하였습니다.

NackUn commented on BeokBeok/app/src/main/java/com/architecturestudy/common/UpBitConnector.kt:
```diff
@@ -0,0 +1,70 @@
+                response: Response<List<UpBitTickerResponse>>
+            ) {
+                val markets = response.body()
+                getTickers(markets?.asSequence()?.filter { prefix.length == 3 || prefix.length == 4 }
```
?.filter { prefix.length == 3 || prefix.length == 4 } 이부분이 없어도 될 것 같습니다.

BeokBeok replied:
> ?.filter { prefix.length == 3 || prefix.length == 4 } 이부분이 없어도 될 것 같습니다.

이 부분은 예외처리를 위해 추가하였습니다.
가령, KRW, USDT와 같이 글자 수로 필터링을 하기 위함입니다.
특정 String으로 비교하도록 수정하겠습니다.

BeokBeok replied:
접두어를 필터링 하도록 함수 추가하였습니다.
f68175f36edeebf6fe35fc6b1402981ffbd3d477

NackUn replied:
확인했습니다. 고생하셨습니다

NackUn commented on BeokBeok/app/src/main/AndroidManifest.xml:
```diff
@@ -1,19 +1,23 @@
             <intent-filter>
-                <action android:name="android.intent.action.MAIN"/>
+                <action android:name="android.intent.action.MAIN" />
+                <action android:name="android.intent.action.VIEW" />
```
<action android:name="android.intent.action.VIEW" />
이건 어떤 이유로 사용하신 건지 궁금합니다.

BeokBeok replied:
<action android:name="android.intent.action.VIEW" />
이 것을 설정하지 않으면, Google Search 시 App이 인덱싱이 되지 않습니다.
해서, 추가하였습니다.
개인적으로 warning(노란색)은 다 수정하는 편입니다.
참고로, 에러는 아닙니다.

NackUn replied:
감사합니다! 덕분에 하나 더 배워갑니다^^ (Resolve 처리하겠습니다)

Commit:
HW-1 Prefix 체크하는 기능 추가 및 retrofit 버전 업데이트

sjjeong commented:
@NackUn 
리뷰가 끝나고 변경점이 더이상 없다고 생각하셔서 Mentor Needed를 추가 하셨다면 approve도 같이 해주셔야 해요

NackUn reviewed (dismissed):
고생하셨습니다!

sjjeong commented on BeokBeok/app/src/main/java/com/architecturestudy/common/UpBitConnector.kt:
```diff
@@ -0,0 +1,74 @@
+                response: Response<List<UpBitTickerResponse>>
+            ) {
+                val markets = response.body()
+                getTickers(markets?.asSequence()?.filter { hasPrefix(prefix) }
```
null chaining으로 처리보다 response.body() null체크를 하셔서 NonNull로 처리하면 좋을 것 같습니다.
getTicker()안에 파라미터로 들어가는 코드가 복잡해서 따로 변수로 선언해주면 좋을 것 같습니다.
https://kotlinlang.org/docs/reference/coding-conventions.html#chained-call-wrapping 참고하셔서 chaining 부분 line break 해주시면 좋을 것 같습니다.

BeokBeok replied:
적용하였습니다.
c52e535

sjjeong commented on BeokBeok/app/src/main/java/com/architecturestudy/common/UpBitConnector.kt:
```diff
@@ -0,0 +1,74 @@
+        return listOf("KRW", "BTC", "ETH", "USDT").contains(prefix)
+    }
+
+    interface Retrofittable {
```
interface 이름을 보고 어떤 것인지 파악하기 어렵습니다.
UpBitApi로 변경하면 어떨까요?

BeokBeok replied:
UpBitService 로 명명하여 반영하였습니다.
d642443

sjjeong commented on BeokBeok/app/src/main/java/com/architecturestudy/model/UpBitTickerResponse.kt:
```diff
@@ -0,0 +1,21 @@
+
+data class UpBitTickerResponse(
+
+    @field:SerializedName("market")
```
`field:`는 생략해도 됩니다.

BeokBeok replied:
생략하였습니다.
7c58f74

sjjeong commented on BeokBeok/app/src/main/java/com/architecturestudy/view/MainFragment.kt:
```diff
@@ -0,0 +1,39 @@
+    }
+
+    override fun onSuccessMarketPrice(marketPrice: List<UpBitTickerResponse>) {
+        val context = activity as MainActivity
```
ui를 setup하는 부분과 data를 set하는 부분을 분리해주세요.

BeokBeok replied:
적용하였습니다.
311bf0d

sjjeong commented on BeokBeok/app/src/main/java/com/architecturestudy/view/MainAdapter.kt:
```diff
@@ -0,0 +1,45 @@
+import com.architecturestudy.model.UpBitTickerResponse
+import java.text.DecimalFormat
+
+class MainAdapter(private val context: Context, private val marketPrice: List<UpBitTickerResponse>?) :
```
Adapter에 생성자로 data를 받는것은 좋지 않습니다.
내부에 list를 만들고 함수를 통해서 데이터를 넣어주는 방식으로 변경해주세요.

외부에서 context를 받아서 사용하지 말아주세요.

BeokBeok replied:
적용하였습니다.
b1b4adc

sjjeong commented on BeokBeok/app/src/main/java/com/architecturestudy/view/MainAdapter.kt:
```diff
@@ -0,0 +1,45 @@
+        marketPrice?.get(position)?.let { holder.bind(it) }
+    }
+
+    inner class ViewHolder(itemView: View?) : RecyclerView.ViewHolder(itemView!!) {
```
itemView를 NonNull로 변경해주세요.

BeokBeok replied:
적용하였습니다.
5a2cf3d

sjjeong commented on BeokBeok/app/src/main/java/com/architecturestudy/view/MainAdapter.kt:
```diff
@@ -0,0 +1,45 @@
+                    marketPrice.market
+            currentPrice?.text = DecimalFormat("#,###").format(marketPrice.tradePrice)
+            netChange?.text = String.format("%.2f", marketPrice.signedChangeRate?.times(100))
+            tradingVal?.text = DecimalFormat("#,###M").format(marketPrice.accTradePrice24h?.div(1000000))
```
1000000보단 1_000_000이 한눈에 보기 더 좋아요

BeokBeok replied:
적용하였습니다.
58ddcd6

sjjeong commented on BeokBeok/app/src/main/res/layout/fragment_main.xml:
```diff
@@ -0,0 +1,83 @@
+            <TextView
+                android:id="@+id/tv_coin_name"
+                android:layout_width="0dp"
+                android:layout_height="match_parent"
```
ConstraintLayout 자식에 있는 View는 match_parent보다 0dp를 권장 합니다.

BeokBeok replied:
@sjjeong 
width와 height 둘 다 0dp를 권장하는 것인가요?

sjjeong replied:
@BeokBeok 
네 match_parent가 동작이 이상해서 0dp를 쓰는게 좋습니다.

BeokBeok replied:
적용하였습니다.
4e797f1

sjjeong replied:
16번째 줄도 수정해주셔야 합니다

BeokBeok replied:
추가 적용하였습니다.
4133ff6

sjjeong reviewed (changes requested):
reformat code도 해서 올려주세요.

BeokBeok committed:
HW-1 response.body() null 체크 및 chained-call-wrapping 적용

- Review URL
https://github.com/StudyFork/GoogryAndroidArchitectureStudy/pull/53#discussion_r292979876

BeokBeok committed:
HW-1 인터페이스명 변경(Retrofittable -> UpBitService)

- Retrofit2 가이드 참고하여 변경
https://square.github.io/retrofit/
- Review URL
https://github.com/StudyFork/GoogryAndroidArchitectureStudy/pull/53#discussion_r292981372

BeokBeok committed:
HW-1 field annotation 제거

- Review URL
https://github.com/StudyFork/GoogryAndroidArchitectureStudy/pull/53#discussion_r292981964

BeokBeok committed:
HW-1 Adapter에 생성자로 context 및 data를 받지 않도록 변경

- Review URL
https://github.com/StudyFork/GoogryAndroidArchitectureStudy/pull/53#discussion_r292986151

BeokBeok committed:
HW-1 ui와 data를 set하는 부분 분리

- Fragment 생명주기에 맞게 ui와 data를 set 하도록 변경

- Review URL
https://github.com/StudyFork/GoogryAndroidArchitectureStudy/pull/53#discussion_r292985574

BeokBeok committed:
HW-1 itemView 를 NonNull로 변경

- Review URL
https://github.com/StudyFork/GoogryAndroidArchitectureStudy/pull/53#discussion_r292986864

BeokBeok committed:
HW-1 숫자 자리수 표기 적용

- Review URL
https://github.com/StudyFork/GoogryAndroidArchitectureStudy/pull/53#discussion_r292987640

BeokBeok committed:
HW-1 ConstraintLayout child view width와 height 0dp 설정

- Review URL
https://github.com/StudyFork/GoogryAndroidArchitectureStudy/pull/53#discussion_r292988995

Commit:
Merge remote-tracking branch 'origin/class02/BeokBeok' into class02/BeokBeok

BeokBeok committed:
HW-1 ConstraintLayout child view width와 height 0dp 설정 추가

- Review URL
https://github.com/StudyFork/GoogryAndroidArchitectureStudy/pull/53#discussion_r292988995

sjjeong reviewed (approved):
LGTM ✊ ✌️ 👋

NackUn reviewed (approved):
고생하셨습니다!
